src/many-kvstore/src/module/event.rs

```diff
@@ -93,3 +99,106 @@ fn filter_date<'a>(
         Ok(events::EventLog { time, .. }) => range.contains(time),
     }))
 }
+
+fn filter_attribute_specific<'a>(
+    mut it: Box<dyn Iterator<Item = EventLogResult> + 'a>,
+    attribute_specific: &'a BTreeMap<
+        EventFilterAttributeSpecificIndex,
+        EventFilterAttributeSpecific,
+    >,
+) -> Box<dyn Iterator<Item = EventLogResult> + 'a> {
+    for x in attribute_specific.values() {
+        match x {
+            EventFilterAttributeSpecific::MultisigTransactionState(VecOrSingle(state)) => {
+                it = Box::new(it.filter(|t| match t {
+                    Err(_) => true,
+                    Ok(EventLog {
+                        content: EventInfo::AccountMultisigSubmit { .. },
+                        ..
+                    })
+                    | Ok(EventLog {
+                        content: EventInfo::AccountMultisigApprove { .. },
+                        ..
+                    }) => state.contains(&MultisigTransactionState::Pending),
+                    Ok(EventLog {
+                        content: EventInfo::AccountMultisigExecute { .. },
+                        ..
+                    }) => {
+                        state.contains(&MultisigTransactionState::ExecutedAutomatically)
+                            || state.contains(&MultisigTransactionState::ExecutedManually)
+                    }
+                    Ok(EventLog {
+                        content: EventInfo::AccountMultisigWithdraw { .. },
+                        ..
+                    }) => state.contains(&MultisigTransactionState::Withdrawn),
+                    Ok(EventLog {
+                        content: EventInfo::AccountMultisigExpired { .. },
+                        ..
+                    }) => state.contains(&MultisigTransactionState::Expired),
+                    _ => false,
+                }))
+            }
+        }
+    }
```

Comment:
FYI, for now, at least, Multisig* has nothing to do with kvstore. 

I don't think we need to support this impl at the moment.


However, you definitely need to add it to `many-ledger`, which is missing.